src/components/shared/Fields/Form/Form.tsx

```diff
@@ -129,6 +129,20 @@ const Form = <FormData extends FieldValues>({
     [handleSubmit, onSubmit, formHelpers, onError],
   );
 
+  // Separate concern for handling defaultValues updates
+  useEffect(() => {
+    const initializeForm = async () => {
+      const resolvedDefaultValues =
+        typeof defaultValues === 'function'
+          ? await defaultValues()
+          : defaultValues;
+
+      formHelpers.reset(resolvedDefaultValues);
+    };
+
+    initializeForm();
+  }, [defaultValues, formHelpers]);
```

Comment:
we could maybe just do `formHelpers.reset` as the dependency, not sure if any other helpers can change according to form state